ch04/src/Solution415.java

```diff
@@ -0,0 +1,30 @@
+import java.util.Scanner;
+
+public class Solution415 {

```

Comment:
Would really benefit from a few comments, however, a truly ingenious solution!

A much more verbose, but maybe easier to read solution would be, to use a `switch` block with appropriate fallthrough. For example:

~~~~Java
switch (letterInUpperCaseNumeric) {
case 'A':
case 'B':
case 'C':
  System.out.println("2");
  break;
case 'D':
case 'E':
case 'F':
  System.out.println("3");
  break;
...
default:
  System.out.println("Invalid input");
}
~~~~